.github/ISSUE_TEMPLATE/2-bug-report.md

```diff
@@ -0,0 +1,86 @@
+---
+name: Bug report
+about: Report a bug to help improve this project
+---
+
+### Checklist
+
+<!-- Please make sure you check all these items before submitting your bug report. -->
+
+- [ ] The bug is reproducible against the latest release and/or `master`.
+- [ ] There are no similar issues or pull requests to fix it yet.
+
+### Describe the bug
+
+<!-- A clear and concise description of what the bug is. -->
+
+### To reproduce
+
+<!-- Provide a *minimal* example with steps to reproduce the bug locally.
+
+If you are requesting an external host, please try to setup a local server that allows reproducing the issue.
+
+An example is provided below, feel free to adapt it to your needs. -->
+
+1. Create and start a local server:
+
+```python
+# $ pip install starlette uvicorn
+import uvicorn
+from starlette import Starlette
+from starlette.routing import Route
+from starlette.responses import PlainTextResponse
+
+async def home(request):
+    return PlainTextResponse("Hello, world!")
+
+app = Starlette(routes=[Route("/", home)])
+
+uvicorn.run(app)
+```
+
+2. Create a test script:
+
+```python
+import httpx
+
+r = httpx.get("http://localhost:8000")
+print(r)
+```
+
+4. Run the test script.
+5. It raises an error.
+6. But I expected it to successfully print the response.
```

Comment:
This is great but I wonder if we should add a link to these instructions instead, maybe in the docs?

I just worry it might be too much information for someone coming in.

Reply:
I'm really not sure where this would be in the docs… Also, having these instructions away from the issue template doesn't sound great in terms of keeping these up to date…

The motivation for providing an example of a repro case is because a lot of the bug reports we get assume that attaching a script is a repro example… skipping the important part that it shouldn't have any external dependencies.

I updated the template to drop the code example, and add a bit more prose with a focus on "keep dependencies at a minimum". :-)

Reply:
Yeah, the docs is probably not the right place for that. 👍